### PR TITLE
Add query timings

### PR DESCRIPTION
**Pull request by iwiznia**
### Details
Was looking into why an API command was slow in [this request](https://www.expensify.com/_devportal/tools/logSearch/?#query=request_id:(%22oYRj33%22)+AND+timestamp:[2023-02-16T12:31:46.102Z+TO+2023-02-16T14:31:46.102Z]&index=logs_expensify-020941), then noticed that some of the times between the logs of ` Report action type CREATED doesn't have a JSON message` was super high (sometimes 100ms) and I am not sure where time is being spent. There are some queries being done in that for, so maybe it's that or maybe is just the time to run the C++ code. In any case, having the timing information of each query will be very useful to identify where the command spent its time and how to optimize it.

I decided to log everything above 1ms to reduce some noise. We could increase that number a bit if we believe this will be too noisy.

### Tests

```
2023-02-16T14:58:52.844601+00:00 expensidev2004 bedrock: eDkLEE ionatan@expensify.com (libstuff.cpp:2644) SQuery [socket10] [info] Query completed (2ms): SELECT a.accountID, a.email, p.name, p.permissions, p.key, p.value, p.created FROM (SELECT accountID as ownerAccountID, name, created, 'read, write, share, own' AS permissions,json.key AS key, json.value AS value FROM nameValuePairs n, JSON_EACH(n.value) json WHERE accountID=2 AND name GLOB 'expensify_policy*'  AND key IN ('name', 'type', 'outputCurrency', 'lastModified', 'avatarURL') UNION SELECT s.ownerAccountID, n.name AS name, n.created AS created, s.permissions AS permissions, json.key AS key, json.value AS value FROM sharedNameValuePairs s INNER JOIN nameValuePairs n ON n.name=s.name, JSON_EACH(n.value) json WHERE s.accountID=2 AND s.name GLOB 'expensify_policy*'  AND key IN ('name', 'type', 'outputCurrency', 'lastModified', 'avatarURL') ) AS p INNER JOIN accounts a ON p.ownerAccountID = a.accountID;
2023-02-16T14:58:54.080688+00:00 expensidev2004 bedrock: aZrbnY ionatan@expensify.com (libstuff.cpp:2644) SQuery [socket11] [info] Query completed (2ms): SELECT name FROM nameValuePairs WHERE accountID = 2 AND name GLOB 'expensify_policy*' AND json_extract(value, '$.type') = 'personal';
2023-02-16T14:58:54.082666+00:00 expensidev2004 bedrock: aZrbnY ionatan@expensify.com (libstuff.cpp:2644) SQuery [socket11] [info] Query completed (1ms): SELECT COUNT(1) FROM sharedNameValuePairs s INNER JOIN accounts sa ON sa.accountID = s.accountID WHERE s.name GLOB 'expensify_policy*' AND s.ownerAccountID = 2 AND s.permissions = 'read, write, share, own' AND ( (SELECT value FROM nameValuePairs WHERE name = 'private_approvedAccountant' AND accountID = s.accountID) = 'APPROVED_ACCOUNTANT' OR (SELECT value FROM nameValuePairs WHERE name = 'private_approvedAccountant' AND accountID = (SELECT accountID FROM accounts WHERE email = '+@' || SUBSTR(SUBSTR(sa.email, INSTR(sa.email, '@') + 1), INSTR(SUBSTR(sa.email, INSTR(sa.email, '@') + 1), '@') + 1))) = 'APPROVED_ACCOUNTANT' );
2023-02-16T14:58:54.087029+00:00 expensidev2004 bedrock: aZrbnY ionatan@expensify.com (libstuff.cpp:2644) SQuery [socket11] [info] Query completed (2ms): SELECT COUNT(*) FROM nameValuePairs WHERE name IN ( 'expensify_policy4221DA4478F819AC', 'expensify_policy67ADBB6952CDB74E', 'expensify_policy7BB40DC797749DEE', 'expensify_policy88A323482C242613', 'expensify_policyA39D0C813DE79ECB', 'expensify_policyADC7BA1EC51327A1', 'expensify_policyB57CBD38AC3A7D0B' ) AND JSON_VALID(value) AND JSON_EXTRACT(value, '$.connections.xero.data.countryCode') = 'AU';
2023-02-16T14:58:54.107309+00:00 expensidev2004 bedrock: aZrbnY ionatan@expensify.com (libstuff.cpp:2644) SQuery [socket11] [info] Query completed (4ms): SELECT a.accountID, a.email, p.name, p.permissions, p.key, p.value, p.created FROM (SELECT accountID as ownerAccountID, name, created, 'read, write, share, own' AS permissions,json.key AS key, json.value AS value FROM nameValuePairs n, JSON_EACH(n.value) json WHERE accountID=2 AND name GLOB 'expensify_policy*'  AND key IN ('name', 'type', 'outputCurrency', 'lastModified', 'avatarURL') UNION SELECT s.ownerAccountID, n.name AS name, n.created AS created, s.permissions AS permissions, json.key AS key, json.value AS value FROM sharedNameValuePairs s INNER JOIN nameValuePairs n ON n.name=s.name, JSON_EACH(n.value) json WHERE s.accountID=2 AND s.name GLOB 'expensify_policy*'  AND key IN ('name', 'type', 'outputCurrency', 'lastModified', 'avatarURL') ) AS p INNER JOIN accounts a ON p.ownerAccountID = a.accountID;
2023-02-16T14:58:54.192324+00:00 expensidev2004 bedrock: xlSrFs ionatan@expensify.com (libstuff.cpp:2644) SQuery [socket12] [info] Query completed (5ms): SELECT a.accountID, a.email, p.name, p.permissions, p.key, p.value, p.created FROM (SELECT accountID as ownerAccountID, name, created, 'read, write, share, own' AS permissions,json.key AS key, json.value AS value FROM nameValuePairs n, JSON_EACH(n.value) json WHERE accountID=2 AND name GLOB 'expensify_policy*'  AND key IN ('name', 'type', 'outputCurrency', 'lastModified', 'avatarURL') UNION SELECT s.ownerAccountID, n.name AS name, n.created AS created, s.permissions AS permissions, json.key AS key, json.value AS value FROM sharedNameValuePairs s INNER JOIN nameValuePairs n ON n.name=s.name, JSON_EACH(n.value) json WHERE s.accountID=2 AND s.name GLOB 'expensify_policy*'  AND key IN ('name', 'type', 'outputCurrency', 'lastModified', 'avatarURL') ) AS p INNER JOIN accounts a ON p.ownerAccountID = a.accountID;
2023-02-16T14:58:54.235127+00:00 expensidev2004 bedrock: rcfRIN ionatan@expensify.com (libstuff.cpp:2644) SQuery [socket13] [info] Query completed (2ms): SELECT a.accountID, a.email, p.name, p.permissions, p.key, p.value, p.created FROM (SELECT accountID as ownerAccountID, name, created, 'read, write, share, own' AS permissions,json.key AS key, json.value AS value FROM nameValuePairs n, JSON_EACH(n.value) json WHERE accountID=2 AND name GLOB 'expensify_policy*'  AND key IN ('name', 'type', 'outputCurrency', 'lastModified', 'avatarURL') UNION SELECT s.ownerAccountID, n.name AS name, n.created AS created, s.permissions AS permissions, json.key AS key, json.value AS value FROM sharedNameValuePairs s INNER JOIN nameValuePairs n ON n.name=s.name, JSON_EACH(n.value) json WHERE s.accountID=2 AND s.name GLOB 'expensify_policy*'  AND key IN ('name', 'type', 'outputCurrency', 'lastModified', 'avatarURL') ) AS p INNER JOIN accounts a ON p.ownerAccountID = a.accountID;
2023-02-16T14:58:54.267040+00:00 expensidev2004 bedrock: qxT5b5 ionatan@expensify.com (libstuff.cpp:2644) SQuery [socket14] [info] Query completed (2ms): SELECT a.accountID, a.email, p.name, p.permissions, p.key, p.value, p.created FROM (SELECT accountID as ownerAccountID, name, created, 'read, write, share, own' AS permissions,json.key AS key, json.value AS value FROM nameValuePairs n, JSON_EACH(n.value) json WHERE accountID=2 AND name GLOB 'expensify_policy*'  AND key IN ('name', 'type', 'outputCurrency', 'lastModified', 'avatarURL') UNION SELECT s.ownerAccountID, n.name AS name, n.created AS created, s.permissions AS permissions, json.key AS key, json.value AS value FROM sharedNameValuePairs s INNER JOIN nameValuePairs n ON n.name=s.name, JSON_EACH(n.value) json WHERE s.accountID=2 AND s.name GLOB 'expensify_policy*'  AND key IN ('name', 'type', 'outputCurrency', 'lastModified', 'avatarURL') ) AS p INNER JOIN accounts a ON p.ownerAccountID = a.accountID;
2023-02-16T14:58:56.817805+00:00 expensidev2004 bedrock: uXTqBP ionatan@expensify.com (libstuff.cpp:2644) SQuery [socket15] [info] Query completed (2ms): SELECT a.accountID, a.email, p.name, p.permissions, p.key, p.value, p.created FROM (SELECT accountID as ownerAccountID, name, created, 'read, write, share, own' AS permissions,json.key AS key, json.value AS value FROM nameValuePairs n, JSON_EACH(n.value) json WHERE accountID=2 AND name GLOB 'expensify_policy*'  AND key IN ('name', 'type', 'outputCurrency', 'lastModified', 'avatarURL') UNION SELECT s.ownerAccountID, n.name AS name, n.created AS created, s.permissions AS permissions, json.key AS key, json.value AS value FROM sharedNameValuePairs s INNER JOIN nameValuePairs n ON n.name=s.name, JSON_EACH(n.value) json WHERE s.accountID=2 AND s.name GLOB 'expensify_policy*'  AND key IN ('name', 'type', 'outputCurrency', 'lastModified', 'avatarURL') ) AS p INNER JOIN accounts a ON p.ownerAccountID = a.accountID;
2023-02-16T14:58:56.886975+00:00 expensidev2004 bedrock: Y1vO9x ionatan@expensify.com (libstuff.cpp:2644) SQuery [socket17] [info] Query completed (2ms): SELECT a.accountID, a.email, p.name, p.permissions, p.key, p.value, p.created FROM (SELECT accountID as ownerAccountID, name, created, 'read, write, share, own' AS permissions,json.key AS key, json.value AS value FROM nameValuePairs n, JSON_EACH(n.value) json WHERE accountID=2 AND name GLOB 'expensify_policy*'  AND key IN ('name', 'type', 'outputCurrency', 'lastModified', 'avatarURL') UNION SELECT s.ownerAccountID, n.name AS name, n.created AS created, s.permissions AS permissions, json.key AS key, json.value AS value FROM sharedNameValuePairs s INNER JOIN nameValuePairs n ON n.name=s.name, JSON_EACH(n.value) json WHERE s.accountID=2 AND s.name GLOB 'expensify_policy*'  AND key IN ('name', 'type', 'outputCurrency', 'lastModified', 'avatarURL') ) AS p INNER JOIN accounts a ON p.ownerAccountID = a.accountID;
```
_________
**Internal Testing Reminder:** when changing bedrock, please compile auth against your new changes

### CODE REVIEW

File: libstuff/libstuff.cpp [hunk 1]
```diff
@@ -2614,15 +2614,14 @@ int SQuery(sqlite3* db, const char* e, const string& sql, SQResult& result, int6
...
+    // This code removing authTokens is a quick fix and should be removed once https://github.com/Expensify/Expensify/issues/144185 is done.
     string sqlToLog = sql;
-    if ((int64_t)elapsed > warnThreshold) {
-        // We should always avoid logging authTokens because they give access to accounts
-        pcrecpp::RE("\"authToken\":\"[0-9A-F]{400,1024}\"").GlobalReplace("\"authToken\":<REDACTED>", &sqlToLog);
-
-        // We remove anything inside "html" because we intentionally don't log chats
-        pcrecpp::RE("\"html\":\".*\"").GlobalReplace("\"html\":\"<REDACTED>\"", &sqlToLog);
+    // We should always avoid logging authTokens because they give access to accounts
+    pcrecpp::RE("\"authToken\":\"[0-9A-F]{400,1024}\"").GlobalReplace("\"authToken\":<REDACTED>", &sqlToLog);
```
Comment (flodnv) [inline libstuff/libstuff.cpp:2620]: I am slightly concerned about running this code always now. Can we at least run it only if we're going to log the query?

File: libstuff/libstuff.cpp [hunk 2]
```diff
@@ -2639,6 +2638,10 @@ int SQuery(sqlite3* db, const char* e, const string& sql, SQResult& result, int6
         if (wasSlow != nullptr) {
             *wasSlow = true;
         }
+    } else if ((int64_t)elapsed > 1000) {
```
Comment (flodnv) [inline libstuff/libstuff.cpp:2641]: I think 10ms would be a better threshold

**Review by flodnv — COMMENTED** [review]
What do you think @tylerkaraszewski ?

**Comment by iwiznia** [comment]
Updated

**Review by tylerkaraszewski — APPROVED** [review]

**Comment by tylerkaraszewski** [comment]
I'm fine with this. I'm not really worried about the time spent in the regex.

**Pull request merged**